pkg/quota/v1/evaluator/core/registry.go

```diff
@@ -32,6 +32,15 @@ var legacyObjectCountAliases = map[schema.GroupVersionResource]corev1.ResourceNa
 	corev1.SchemeGroupVersion.WithResource("secrets"):                corev1.ResourceSecrets,
 }
 
+func NeedFullObjectForResource(resource schema.GroupVersionResource) bool {
+	switch resource.GroupResource() {
+	case schema.GroupResource{Resource: "pods"}, schema.GroupResource{Resource: "services"}, schema.GroupResource{Resource: "persistentvolumeclaims"}:
+		return true
+	default:
+		return false
+	}
+}
+
```

Comment:
can this be included in the `quota.Evaluator` interface so podEvaluator / serviceEvaluator / persistentVolumeClaim evaluator answer whether it requires the full object or not, and we don't ever drift?

Reply:
The evaluator might be created after this function (see pkg/controller/resourcequota/resource_quota_monitor.go:controllerFor() ), it seems infeasible to me. At least not without major changes.